### ISSUE TITLE
Restore the "duration" function to node.sleep()

### ISSUE THREAD

**Issue by mikep01** [nodemcu/nodemcu-firmware#2442]
8<------------------------ FEATURE REQUEST ------------------------------------
### Missing feature
In 2.1 node.sleep() had a "duration" parameter that would awaken the CPU after the given period.
This parameter and function has been removed from 2.2?

### Justification
Many battery powered IoT devices need to use light sleep, waking for a periodic check in with a controller.  

### Workarounds
None.  Node.dsleep requires hardware changes and a full restart (too slow).

8<------------------------ END FEATURE REQUEST --------------------------------

**Comment by mikep01** [comment]
Reviewing the timer suspend refactoring it appears that the developer gave up and removed the node.sleep duration feature.

This feature was indeed broken in the ESP SDK 2.1 release.  It was fixed and was working properly with NodeMCU 2.1 after I manually updated the SDK to 2.2.  Perhaps this is the reason the developer couldn't get it to work?

**Comment by mikep01** [comment]
I back (forward?) ported the old timer function into NodeMCU 2.2 dev and the node.sleep(duration=xxx) works just fine.

FWIW, tmr.suspend_all and tmr.resume_all are working too.  These functions are useful when interrupting init.lua from an IDE if it's using a tmr to pause (which mine does, I have a "pause" button).

Also, there's something funky about the timer enable defines in user_config.h.  Before I started back porting I found enabling them wasn't quite working - both the old and new defines were needed.  Probably worth a look.
